src/c++/simulator/analitic_geometry.hpp

```diff
@@ -0,0 +1,33 @@
+﻿#pragma once
+
+#include <simulator/boundaries.hpp>
+#include <simulator/particle.hpp>
+#include <simulator/my_types.hpp>
+#include <simulator/geometry.hpp>
+
+#include <iostream>
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+namespace geometric
+{
+	double distance(SphericParticle* p1, SphericParticle* p2);
+	double distance(SphericParticle* p1, Wall* b1);
+	// double distance(Boundary* b1, SphericParticle* p1);
+	double distance(Point* p0, Point* p1);
+	double distance(Point& p0, Point& p1);
```

Comment:
I think we should keep only the `&` version. Since the calculations don't change the values of the `Point`s I would prefer use `Point Const&`